### PR TITLE
ENH: use mamba

### LINKED ISSUE #67: Investigate use of mamba and mambabuild for CI conda builds
See
https://github.com/mamba-org/mamba
https://github.com/mamba-org/boa

### PR DESCRIPTION
**Pull request by ZLLentz**
closes #67 if it works
I'll pick a repo to test this on, probably typhos because it runs the benchmark build even though we don't really look at the output

### CODE REVIEW

**Comment by ZLLentz** [comment]
The linked typhos PR passes the CI now- I'm going to spend a bit of time looking over the output execution at https://app.travis-ci.com/github/pcdshub/typhos/builds/251078960

**Comment by ZLLentz** [comment]
It seems like this is a dramatic speedup.

Timings before:
image

Timings after:
image

**Comment by ZLLentz** [comment]
The python 3.7 builds in particular (the one that's failing right now on pcdsdaq) were really starting to slog and have huge gains in the before timings

**Comment by ZLLentz** [comment]
Note: I did not touch the anaconda upload at all- should I? It only takes about a minute to run and I was scared of breaking it and not being able to easily test the failure. But maybe now that the rest is working it's worth doing that one too.

File: travis/shared_configs/docs-build.yml [hunk 1]
```diff
@@ -14,24 +14,28 @@ jobs:
...
-        - export PATH="$HOME/miniconda/bin:$PATH"
-        - conda config --set always_yes yes --set changeps1 no
-        - conda config --set channel_priority strict
-        - conda config --remove channels defaults
-        - conda config --add channels pcds-tag
-        - conda config --add channels conda-forge
-        - conda config --add channels "file://`pwd`/bld-dir"
+        - curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/micromamba
+        - eval "$(./bin/micromamba shell hook --shell=bash)"
+        - mkdir ~/micromamba
```
Comment (klauer) [inline travis/shared_configs/docs-build.yml:19]: What's this for?

**Review by klauer — APPROVED** [review]
The proof is in the pudding here, and the tasty pudding gets delivered faster than before.

This will be so nice for our development efforts! Thank you! 🥳 

> Note: I did not touch the anaconda upload at all- should I? It only takes about a minute to run and I was scared of breaking it and not being able to easily test the failure. But maybe now that the rest is working it's worth doing that one too.

I think it's fine to leave this alone for now.

File: travis/shared_configs/docs-build.yml (hunk 1 above)
Comment (ZLLentz) [reply travis/shared_configs/docs-build.yml:19]: micromamba needs a target directory akin to ~/miniconda3 to store the environments it creates. The default path is ~/micromamba and it does not create the directory itself. There must be a built-in command to initialize this but I didn't figure out all the details.

Comment (klauer) [reply travis/shared_configs/docs-build.yml:19]: I see - that makes sense. It may be worth adding a comment to that effect.

**Review by tangkong — APPROVED** [review]
Speed improvements look great to me.  Excited to see this merged.

File: travis/shared_configs/docs-build.yml (hunk 1 above)
Comment (ZLLentz) [reply travis/shared_configs/docs-build.yml:19]: Easy to add a few comments :+1:

**Pull request merged**